src/artifact/tar/platform/libarchive/wrapper.cpp

```diff
@@ -81,11 +81,12 @@ Error Handle::Init() {
 	}
 	r = archive_read_open(archive_.get(), &reader_container_, nullptr, reader_callback, nullptr);
 	if (r != ARCHIVE_OK) {
-		return MakeError(
-			error::GenericError,
-			"Failed to initalize the 'libarchive' C bindings. LibArchive error message: '"
-				+ string(archive_error_string(archive_.get()))
-				+ "' error code: " + std::to_string(archive_errno(archive_.get())));
+		string err_msg = "Failed to initialize the 'libarchive' C bindings. LibArchive error code: "
+						 + std::to_string(archive_errno(archive_.get()));
```

Comment:
According to the docs:
>  On error, the `read` callback should invoke archive_set_error() to register an error code and message

IOW, we should be setting the error. Also, the log message is misleading, this is not an initialization of the C bindings, it's actually opening the archive.